nni/retiarii/nn/pytorch/utils.py

```diff
@@ -1,6 +1,35 @@
-from typing import Any, Optional, Tuple
+from abc import ABC, abstractclassmethod
+from typing import Any, Optional, Tuple, Union
 
-from nni.retiarii.utils import ModelNamespace, get_current_context
+import torch.nn as nn
+from nni.retiarii.utils import NoContextError, ModelNamespace, get_current_context
+
+
+class Mutable(ABC, nn.Module):
```

Comment:
I'm not sure but to my understanding `ABC` is not designed for this scenario. The PEP states it's mainly for overriding `isinstance`, and I believe we don't want to treat all objects with `create_fixed_module` method as mutable. To force subclass overriding a virtual method, `NotImplementedError` should be enough.

Anyway, I guess inheriting `ABC` won't cause problem.

Reply:
I might want to refine and make `Mutable` a more clear interface, which defines what a "high-level API" should do.

For example, each "mutable" class should have a corresponding mutator, and a dedicated approach to preprocess the "mutable" into base class + mutator. For graph engine, this might even include the part where the model is parsed into graph.

On the other hand, I concern that such design is over complex, and we should have another "Adapter" module that handles all the complex logics there.

I haven't made the decision yet. Let's keep it for now and I'd like to have more thoughts about it.

Reply:
Seems we are talking about different things...
I mean, although it's an "abstract base class" (in English), it does not need to be `ABC` (in Python).
To my understanding inheriting `ABC` implies that the user does not need to inherit this class. Instead they only need to provide required methods.
For example a class with `__iter__` method is considered `issubclass(MyContainer, Iterable)`, even though it does not actually inherit `Iterable`.

Reply:
Read the PEP and some comments and SO, seems that I'm abusing ABC.
I'll consider removing it.